gtk3/sample/misc/flowbox.rb

```diff
@@ -46,37 +46,36 @@ def initialize
     scrolled = Gtk::ScrolledWindow.new
     scrolled.set_policy(Gtk::PolicyType::NEVER, Gtk::PolicyType::AUTOMATIC)
 
-    @flowbox = Gtk::FlowBox.new
-    @flowbox.set_valign(Gtk::Align::START)
-    @flowbox.set_max_children_per_line(30)
-    @flowbox.set_selection_mode(Gtk::SelectionMode::NONE)
-    fill_flowbox
+    flowbox = Gtk::FlowBox.new
+    flowbox.set_valign(Gtk::Align::START)
+    flowbox.set_max_children_per_line(30)
+    flowbox.set_selection_mode(Gtk::SelectionMode::NONE)
 
-    scrolled.add(@flowbox)
+    COLORS.each do |color|
+      swatch = create_color_swatch(color)
+      flowbox.add swatch
+    end
+
+    scrolled.add(flowbox)
     add(scrolled)
 
     signal_connect("destroy") { Gtk.main_quit }
   end
 
   private
 
-  def color_swatch_new(color_name)
-    color = Gdk::RGBA.parse(color_name)
+  def create_color_swatch(color_name)
     button = Gtk::Button.new
 
     area = Gtk::DrawingArea.new
     area.set_size_request(24, 24)
-    area.override_background_color(0, color)
+    area.signal_connect("draw") do |_, context|
+      color = Gdk::RGBA.parse(color_name)
```

Comment:
It's better that we parse color name outside of the signal handler. Because signal handler is called multiple times but `create_color_swatch` is called only once.

Reply:
all right. Move this line up.